gemhardware/include/gem/hw/optohybrid/HwOptoHybrid.h

```diff
@@ -874,325 +339,175 @@ namespace gem {
-            //uint32_t src = readReg(getDeviceBaseNode(),regName.str()+".CLOCKING.CDCE.SOURCE");
-            //uint32_t flb = readReg(getDeviceBaseNode(),regName.str()+".CLOCKING.CDCE.FALLBACK");
-            ////maybe do a check to ensure that the value has been read properly?
-            return std::make_pair(0,0);
-          };
-
-          ///** Read the VFAT clock source
-          // * @retval bool true uses the external clock, false uses the onboard clock
-          // */
-          //bool getVFATClockSource();
-          //
-          //
-          ///** Set the VFAT onboard clock fallback
-          // * Allows the clock to fall back to
-          // * @param bool true allows the optohybrid to switch back to the onboard clock
-          // * if it fails to lock
-          // */
-          //void setVFATFallback(bool);
-          //
-          ///** Read the VFAT clock fallback status
-          // * @retval returns status of VFAT clock fallback (false does not allow automatic
-          // * fallback to onboard clock
-          // */
-          //bool getVFATFallback();
-          //
-          //
-          ///** Set the CDCE clock source
-          // * @param bool true uses the external clock, false uses the onboard clock
-          // */
-          //void setCDCEClockSource(bool);
-          //
-          ///** Read the CDCE clock source
-          // * @retval bool true uses the external clock, false uses the onboard clock
-          // */
-          //bool getCDCEClockSource();
-          //
-          ///** Set the CDCE clock fallback mode
-          // * @param bool true allows the optohybrid to switch back to the onboard clock if the
-          // * CDCE doesn't lock
-          // */
-          //void setCDCEFallback(bool);
-          //
-          ///** Read the CDCE clock fallback mode
-          // * @retval bool false does not allow the optohybrid to switch the clock back
-          // */
-          //bool getCDCEFallback();
+          /** @} */ // end of ttccounters
 
           ////// pertaining to VFATs \\\\\\*
 
           /**
-           * Sends a read request to all (un-masked) VFATs on the same register
+           * @brief Sends a read request to all (un-masked) VFATs on the same register
            * @param std::string name name of the register to broadcast the request to
            * @param uint32_t mask specifying which VFATs will receive the broadcast command
            * @param bool reset specifying whether to reset the firmware module first
            * @returns a std::vector of uint32_t words, one response for each VFAT
            */
           std::vector<uint32_t> broadcastRead(std::string const& name,
                                               uint32_t    const& mask=gem::hw::utils::ALL_VFATS_BCAST_MASK,
-                                              bool               reset=false);
+                                              bool        const& reset=false);
```

Comment:
I did it for uniformity of intent, with a consideration that a future cleanup may pass by value in the declaration, then in the implementation the signature would be by `const` value, just for the compiler check.
In any event, probably the compiler is smart enough to optimize this...

Reply:
> In any event, probably the compiler is smart enough to optimize this...

It can't if it can't inline, because the reference is part of the binary interface. To confirm, I dumped the x86_64 assembly for the following code with and without `-DREFERENCE` (`gcc` 7.4.0 at `-O3`):
```C++
#ifdef REFERENCE
[[gnu::noinline]] int f(const bool &x)
#else
[[gnu::noinline]] int f(bool x)
#endif
{
    return x ? 42 : 24;
}
```
Without reference:
```asm
_Z1fRKb:
.LFB30:
        .cfi_startproc
        cmp     BYTE PTR [rdi], 0 ; Looks at the LSB of the value pointed to by register RDI
        mov     edx, 42
        mov     eax, 24
        cmovne  eax, edx 
        ret 
        .cfi_endproc
```
With reference:
```asm
_Z1fb:
.LFB30:
        .cfi_startproc
        cmp     dil, 1 ; Direct access to the LSB of register SI (aka DIL)
        sbb     eax, eax 
        and     eax, -18 
        add     eax, 42
        ret 
        .cfi_endproc
```
While I'm quite sure that the access would be cached in L1 memory, it is still a tiny bit slower.